CNV-70244: Update CPU model docs (#111302)
openshift/openshift-docs: modules/virt-about-host-model.adoc

// Module included in the following assemblies:
//
// * virt/managing_vms/cpu_models/virt-schedule-vms.adoc

:_mod-docs-content-type: CONCEPT
[id="virt-about-host-model_{context}"]
= Host-model CPU

[role="_abstract"]

> `host-model` is the default CPU configuration in {VirtProductName}. When a virtual machine instance (VMI) is created with `host-model`, libvirt automatically selects a CPU model that is closest to the physical CPU of the node where the VMI first runs. The CPU features present on this initial node are included in the VMI's domain definition.

Reviewer: ```suggestion
`host-model` is the default CPU configuration in {VirtProductName}. When a virtual machine instance (VMI) is created with `host-model`, libvirt automatically selects a CPU model that is closest to the physical CPU of the node where the VMI first runs. The host model required CPU features present on this initial node are included in the VMI's domain definition.
```

> The CPU model chosen by `host-model` is often not a standard named model supported by libvirt. This is because the initial node may not include all CPU features required by that named model. Libvirt disables any missing features on the initial node, creating a CPU definition that only includes the available features. This approach ensures:

Reviewer 2: ```suggestion
The CPU model chosen by `host-model` is often not a standard named model supported by libvirt, which may  include CPU features that were not on the initial node. Libvirt creates a CPU definition that only includes the available features on the initial node. This approach ensures:
```

* The VMI can start successfully on the initial node.
* Future live migrations only require the subset of CPU features used in the host-model, even if target nodes have different or additional features.

[id="virt-cpu-model-homogeneous-clusters_{context}"]
== Homogeneous clusters

In a homogeneous cluster, all nodes have similar CPU features. The `host-model` CPU model is recommended for homogeneous clusters, and provides the following benefits:

* Provides VMs with the closest match to the physical CPU.
* VMs can use all CPU features available on the nodes.
* Live migration works smoothly because all nodes share the same CPU features.

[id="live-migration-with-host-model_{context}"]
== Live migration with host-model

> Since the CPU model and features are determined by the initial node, all CPU features exposed to the VMI must also exist on any future live migration target nodes.

Reviewer: ```suggestion
Since the CPU model and features are determined by the initial node, all The host model required CPU features exposed to the VMI must also exist on any future live migration target nodes.
```

{VirtProductName} ensures compatibility by:

> . Detecting all `cpu-feature.node.kubevirt.io/<cpuFeature>` labels on the initial node.

Reviewer: ```suggestion
. Detecting all `host-model-required-features.node.kubevirt.io/<cpuFeature>` labels on the initial node.
```

. Storing these labels per VMI.

> . Using these same labels as node selectors for all future live migrations.

Reviewer: ```suggestion
. Using these same required CPU features to set node selectors for all future live migrations.
```
